ruby/neo4j/driver/internal/messaging/common/common_value_packer.rb

```diff
@@ -0,0 +1,186 @@
+module Neo4j::Driver
+  module Internal
+    module Messaging
+      module Common
+        class CommonValuePacker
+           DATE = 'D'
+           DATE_STRUCT_SIZE = 1
+           TIME = 'T'
+           TIME_STRUCT_SIZE = 2
+           LOCAL_TIME = 't'
+           LOCAL_TIME_STRUCT_SIZE = 1
+           LOCAL_DATE_TIME = 'd'
+           LOCAL_DATE_TIME_STRUCT_SIZE = 2
+           DATE_TIME_WITH_ZONE_OFFSET = 'F'
+           DATE_TIME_WITH_ZONE_ID = 'f'
+           DATE_TIME_STRUCT_SIZE = 3
+           DURATION = 'E'
+           DURATION_TIME_STRUCT_SIZE = 4
+           POINT_2D_STRUCT_TYPE = 'X'
+           POINT_2D_STRUCT_SIZE = 3
+           POINT_3D_STRUCT_TYPE = 'Y'
+           POINT_3D_STRUCT_SIZE = 4
+
+          attr_reader :packer
+
+          delegate :pack_struct_header, to: :packer
+
+          def initialize(output)
+            @packer = Packstream::PackStream::Packer.new(output)
+          end
+
+          def pack(value)
+            case value
+            when String
+              packer.pack(value)
+            when Value
+              if value.instance_of? Value::InternalValue
+                pack_internal_value(value)
+              else
+                raise java.lang.IllegalArgumentException, "Unable to pack: #{value}"
+              end
+            when Hash
+              return packer.pack_map_header(0) if value.nil? || value.empty?
+
+              packer.pack_map_header(value.size)
+
+              value.each do |key, value|
+                packer.pack(key)
+                pack(value)
+              end
+            end
+          end
+
+          def pack_internal_value(value)
+            case value.type_constructor
+            when DATE
+              pack_date(value.as_local_date)
+            when TIME
+              pack_time(value.as_offset_time)
+            when LOCAL_TIME
```

Comment:
Hello @klobuczek 
I have converted primitives data types into ruby but didn't invest much time in these types of data types,
but please let me know if needed to handle them too?

Reply:
yes, this part is good